### PR TITLE
feat: 381 blend & gaussian noise

### LINKED ISSUE #381: Add gaussian noise
### Is your feature request related to a problem?

A popular way to train various kinds of generative neural networks is to add gaussian noise to an image and have them restore the original.

### Desired solution

* Method `add_gaussian_noise` on `Image`
* Parameter `standard_deviation` to adjust the standard deviation of the noise
* Raise error if `standard_deviation` is below 0

Possible implementation:
* Create a new image with the same dimensions as the original containing only the noise:
```py
from PIL import Image as PILImage

img_noise = PILImage.effect_noise((image.width, image.height), standard_deviation)
```
* Blend it with the original image using `Image.blend`

### Possible alternatives (optional)

_No response_

### Screenshots (optional)

_No response_

### Additional Context (optional)

_No response_

### PR DESCRIPTION
**Pull request by patrikguempel**
Closes #381 

### Summary of Changes

Images can now be modified with "gaussian noise". Added blend method in the process.

### CODE REVIEW

File: src/safeds/data/image/containers/_image.py [hunk 1]
```diff
@@ -305,6 +305,49 @@ def adjust_brightness(self, factor: float) -> Image:
...
+        if opacity == 0:
+            warnings.warn(
+                "Opacity is 0, this will not make changes to the image.",
+                UserWarning,
+                stacklevel=2,
+            )
+
+        image_copy = copy.deepcopy(self)
+        img_noise = PIL.Image.effect_noise((image_copy.width, image_copy.height), standard_deviation)
+        image_copy._image = PIL.Image.blend(image_copy._image.convert("L"), img_noise.convert("L"), 0.5)
```
Comment (lars-reimann) [inline src/safeds/data/image/containers/_image.py:324]: `opacity` is never used here.

File: src/safeds/data/image/containers/_image.py [hunk 2]
```diff
@@ -305,6 +305,49 @@ def adjust_brightness(self, factor: float) -> Image:
...
+        return image_copy
+
+    def blend(self, other: Image, alpha: float = 0.5) -> Image:
```
Comment (lars-reimann) [inline src/safeds/data/image/containers/_image.py:327]: I don't like `alpha` here:
1. For `add_gaussian_noise` we control opacity, here we control transparency.
2. It's not entirely clear whether it affects the transparency of the original image or the transparency of the other image.

I'd suggest to
1. Configure the opacity here for the sake of consistency (rename `alpha` to `opacity`)
2. Rename the method to something like `add_image` or `overlay_image`. I think `add_image` might be the best choice since it's consistent with `add_gaussian_noise` and because there are other ways to combine images, like subtracting one or multiplying them.

File: src/safeds/data/image/containers/_image.py (hunk 1 above)
Comment (lars-reimann) [reply src/safeds/data/image/containers/_image.py:324]: I would also not expect the image to be converted to greyscale.

File: src/safeds/data/image/containers/_image.py (hunk 2 above)
Comment (lars-reimann) [reply src/safeds/data/image/containers/_image.py:327]: Actually, scratch that. I don't see a major use-case for a method that interpolates between two  images anyway. Please remove the method.

File: src/safeds/data/image/containers/_image.py (hunk 1 above)
Comment (patrikguempel) [reply src/safeds/data/image/containers/_image.py:324]: this is fixed now. had to do some kind of coversion for the blend method to work, now i went with an rgb conversion

File: src/safeds/data/image/containers/_image.py [hunk 3]
```diff
@@ -305,6 +307,25 @@ def adjust_brightness(self, factor: float) -> Image:
         image_copy._image = ImageEnhance.Brightness(image_copy._image).enhance(factor)
         return image_copy
 
+    def add_gaussian_noise(self, standard_deviation: float) -> Image:
```
Comment (Marsmaennchen221) [inline src/safeds/data/image/containers/_image.py:310]: Please add docstrings to this method

**Review by Marsmaennchen221 — CHANGES_REQUESTED** [review]

File: src/safeds/data/image/containers/_image.py [hunk 4]
```diff
@@ -306,6 +308,38 @@ def adjust_brightness(self, factor: float) -> Image:
...
+        """
+        Add Gaussian noise to the image.
+
+        Parameters
+        ----------
+        standard_deviation : float
+            The standard deviation of the Gaussian distribution.
```
Comment (robmeth) [inline src/safeds/data/image/containers/_image.py:318]: you could add that it has to be bigger than or equal 0

File: src/safeds/data/image/containers/_image.py [hunk 5]
```diff
@@ -306,6 +308,38 @@ def adjust_brightness(self, factor: float) -> Image:
...
+
+        Returns
+        -------
+        result : Image
+            The image with added Gaussian noise.
+        """
+        if standard_deviation < 0:
+            raise ValueError("Standard deviation has to be 0 or bigger.")
```
Comment (robmeth) [inline src/safeds/data/image/containers/_image.py:326]: I think `OutOfBoundsError(standard_deviation , name="standard_deviation", lower_bound=ClosedBound(0))` would be more fitting

File: tests/safeds/data/image/containers/test_image.py [hunk 6]
```diff
@@ -384,6 +384,33 @@ def test_should_warn(self, image: Image, factor: float) -> None:
...
+
+        assert image == expected
+
+    @pytest.mark.parametrize(
+        ("image", "standard_deviation"),
+        [(Image.from_png_file(resolve_resource_path("image/boy.png")), -1)],
+        ids=["sigma below zero"],
+    )
+    def test_should_raise_standard_deviation(self, image: Image, standard_deviation: float) -> None:
+        with pytest.raises(ValueError, match="Standard deviation has to be 0 or bigger."):
```
Comment (robmeth) [inline tests/safeds/data/image/containers/test_image.py:410]: adjust to `OutOfBoundError` if changed in `add_gaussian_noise`

File: tests/safeds/data/image/containers/test_image.py [hunk 7]
```diff
@@ -384,6 +384,33 @@ def test_should_warn(self, image: Image, factor: float) -> None:
...
 
 
+class TestAddGaussianNoise:
+    @pytest.mark.parametrize(
+        ("image", "standard_deviation"),
+        [
+            (Image.from_png_file(resolve_resource_path("image/boy.png")), 0.7),
+            (Image.from_png_file(resolve_resource_path("image/boy.png")), 2.5),
+        ],
+        ids=["one", "two"],
```
Comment (robmeth) [inline tests/safeds/data/image/containers/test_image.py:394]: maybe give it better fitting ids like 0.7 and 2.5?

**Review by Marsmaennchen221 — APPROVED** [review]

**Comment by zzril** [comment]
Need to add the new method to the image tutorial also.

Or do we generally create new issues for that? @lars-reimann

**Comment by lars-reimann** [comment]
> Need to add the new method to the image tutorial also.
> 
> Or do we generally create new issues for that? @lars-reimann

No, that should be added as part of this PR.

**Comment by lars-reimann** [comment]
It's now added to the "Image Processing" tutorial.

**Comment by zzril** [comment]
Tests fail locally for me (`top_level_collect`).

**Comment by lars-reimann** [comment]
Run `poetry install`.

**Comment by zzril** [comment]
> Run `poetry install`.
Ofc, thank.s

**Review by zzril — APPROVED** [review]
LGTM.

(Small remark about the tests: I think it would generally make sense to always include the edge cases (like giving a `standard_deviation` of 0) in the test cases.
But we're not doing that consistently anyway, so I'll still accept. Might be sthg for a new issue.)

**Comment by lars-reimann** [comment]
Thanks, I've added that edge case too.

**Review by zzril — APPROVED** [review]
[LGTM](https://lgtm.party/p/ES142H5O/)
